lib/msf/ui/console/module_argument_parsing.rb

```diff
@@ -0,0 +1,200 @@
+# -*- coding: binary -*-
+
+require 'addressable'
+require 'msf/ui/console/command_dispatcher'
+
+module Msf
+module Ui
+module Console
+
+###
+#
+# A centralized mixin to ensure that options are consistently parsed across all module types
+# when running a module's cmd_run/cmd_check/cmd_exploit arguments
+#
+###
+module ModuleArgumentParsing
+
+  # Options which are standard and predictable across all modules
+  @@module_opts = Rex::Parser::Arguments.new(
+    '-h' => [ false, 'Help banner.'                                          ],
+    '-j' => [ false, 'Run in the context of a job.'                          ],
+    '-J' => [ false, 'Force running in the foreground, even if passive.'     ],
+    '-o' => [ true,  'A comma separated list of options in VAR=VAL format.'  ],
+    '-q' => [ false, 'Run the module in quiet mode with no output'           ]
+  )
+
+  @@module_opts_with_action_support = Rex::Parser::Arguments.new(@@module_opts.fmt.merge(
+    '-a' => [ true, 'The action to use. If none is specified, ACTION is used.']
+  ))
+
+  @@exploit_opts = Rex::Parser::Arguments.new(@@module_opts.fmt.merge(
+    '-e' => [ true,  'The payload encoder to use.  If none is specified, ENCODER is used.' ],
+    '-f' => [ false, 'Force the exploit to run regardless of the value of MinimumRank.'    ],
+    '-n' => [ true,  'The NOP generator to use.  If none is specified, NOP is used.'       ],
+    '-p' => [ true,  'The payload to use.  If none is specified, PAYLOAD is used.'         ],
+    '-t' => [ true,  'The target index to use.  If none is specified, TARGET is used.'     ],
+    '-z' => [ false, 'Do not interact with the session after successful exploitation.'     ]
+  ))
+
+  def parse_check_opts(args)
+    help_cmd = proc do |_result|
+      cmd_check_help
+    end
+    parse_opts(@@module_opts_with_action_support, args, help_cmd: help_cmd)&.slice(:datastore_options)
+  end
+
+  def parse_run_opts(args, action: nil)
+    help_cmd = proc do |result|
+      if result[:action].nil?
+        cmd_run_help
+      else
+        cmd_action_help(action)
+      end
+    end
+
+    parse_opts(@@module_opts_with_action_support, args, help_cmd: help_cmd)
+  end
+
+  def parse_exploit_opts(args)
+    help_cmd = proc do |_result|
+      cmd_exploit_help
+    end
+    parse_opts(@@exploit_opts, args, help_cmd: help_cmd)&.except(:action)
+  end
+
+  def print_module_run_or_check_usage(command:, description: nil, options: @@module_opts)
+    description ||= command == :check ? 'Check if the target is vulnerable' : "Run the current #{name.downcase} module"
+
+    is_http_mod = mod.is_a?(Msf::Exploit::Remote::HttpClient)
+    is_smb_mod = mod.is_a?(Msf::Exploit::Remote::SMB::Client) || mod.options.include?('SMBUser')
+    is_mysql_mod = mod.is_a?(Msf::Exploit::Remote::MYSQL)
```

Comment:
This is just a placeholder implementation, we'll probably want to iterate on this with a better long-term solution.
Either way, better usage messages for users is a win in my book